schema.ts

```diff
@@ -1232,6 +1232,80 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
   // TODO(dweitzman): Only support options in the typing when the codaType is ValueHintType.SelectList.
 }
 
+/**
+ * Discriminent used to determine the permission type
+ * @hidden
+ */
+export enum PermissionType {
```

Comment:
Should we be a bit more technical/accurate here and call these principal types?

Reply:
Yes